src/perfrecord.cpp

```diff
@@ -190,3 +224,40 @@ void PerfRecord::sendInput(const QByteArray& input)
     Q_ASSERT(m_perfRecordProcess);
     m_perfRecordProcess->write(input);
 }
+
+bool PerfRecord::checkFilePermissions(const QString &filePath)
+{
+    QFileInfo outputFile(filePath);
+    QString username = currentUsername();
+
+    if (outputFile.owner() != username) {
+        QString sudoExe = sudoUtil();
+        if (sudoExe.isEmpty() || username.isEmpty()) {
+            return false;
+        }
+
+        QString chownCommand = sudoExe + QStringLiteral(" -u root chown ") +
+                               username + QLatin1Char(':') +
+                               username + QLatin1Char(' ') +
+                               filePath;
+        m_chownProcess->start(chownCommand);
+        m_chownProcess->waitForFinished();
+    }
+    return true;
+}
+
+QString PerfRecord::sudoUtil() const
+{
+    QString util = QStandardPaths::findExecutable(QStringLiteral("kdesudo"));
+
+    if (util.isEmpty()) {
+        util = QStandardPaths::findExecutable(QStringLiteral("gksudo"));
+    }
+    return util;
+}
+
+QString PerfRecord::currentUsername() const
```

Comment:
dito

Reply:
This function is used by recordpage.cpp as well. I originally had it in recordutils.h but moved it here after your last review. It's used on line 316

```if (m_perfRecord->currentUsername() == QStringLiteral("root")) {
        ui->recordAsSudoCheckBox->setChecked(true);
        ui->recordAsSudoCheckBox->setEnabled(false);
    }

Reply:
ok